File: Backend/SmartShoppingAssistantLigaAc.DataAccess/Entities/Banner.cs
namespace SmartShoppingAssistantLigaAc.DataAccess.Entities;

public class Banner
{
    public int Id { get; set; }
    public string Title { get; set; } = null!;
    public string? Subtitle { get; set; }
    public string? ImageUrl { get; set; }
    public string? LinkTo { get; set; }
    public int? PromotionId { get; set; }
    public bool IsActive { get; set; }
    public int DisplayOrder { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Promotion? Promotion { get; set; }
}

File: Frontend/smart-shopping-assistant-frontend/src/components/shared/types/Banner.ts
import type { BannerModel } from "../../../api/models/BannerModel";

export interface Banner {
  id: number;
  title: string;
  subtitle: string | null;
  imageUrl: string | null;
  linkTo: string | null;
  promotionId: number | null;
  isActive: boolean;
  displayOrder: number;
  createdAt: string;
}

export function toBanner(dto: BannerModel): Banner {
  return {
    id: dto.id,
    title: dto.title,
    subtitle: dto.subtitle ?? null,
    imageUrl: dto.imageUrl ?? null,
    linkTo: dto.linkTo ?? null,
    promotionId: dto.promotionId ?? null,
    isActive: dto.isActive,
    displayOrder: dto.displayOrder,
    createdAt: dto.createdAt,
  };
}
